homeassistant/components/plex/models.py

```diff
@@ -0,0 +1,126 @@
+"""Models to represent various Plex objects used in the integration."""
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_MOVIE,
+    MEDIA_TYPE_MUSIC,
+    MEDIA_TYPE_TVSHOW,
+    MEDIA_TYPE_VIDEO,
+)
+from homeassistant.util import dt as dt_util
+
+LIVE_TV_SECTION = "-4"
+
+
+class PlexSession:
+    """Represents a Plex playback session."""
+
+    def __init__(self, plex_server, session):
+        """Initialize the object."""
+        self.plex_server = plex_server
+
+        # Available on both media and session objects
+        self.media_content_id = None
+        self.media_content_type = None
+        self.media_content_rating = None
+        self.media_duration = None
+        self.media_image_url = None
+        self.media_library_title = None
+        self.media_summary = None
+        self.media_title = None
+        # TV Shows
+        self.media_episode = None
+        self.media_season = None
+        self.media_series_title = None
+        # Music
+        self.media_album_name = None
+        self.media_album_artist = None
+        self.media_artist = None
+        self.media_track = None
+
+        # Only available on sessions
+        self.player = next(iter(session.players), None)
+        self.device_product = self.player.product
+        self.media_position = session.viewOffset
+        self.session_key = session.sessionKey
+        self.state = self.player.state
+        self.username = next(iter(session.usernames), None)
+
+        # Used by sensor entity
+        sensor_user_list = [self.username, self.device_product]
+        self.sensor_title = None
+        self.sensor_user = " - ".join(filter(None, sensor_user_list))
+
+        self.update_media(session)
+
+    def __repr__(self):
+        """Return representation of the session."""
+        return f"<{self.session_key}:{self.sensor_title}>"
+
+    def update_media(self, media):
```

Comment:
Adding types would really help with understanding this code for newcomers.
I believe `media` here is a `plexapi.audio.Audio | plexapi.video.Video`, right? (I don't know how to do union types in python, but that would be the syntax for typescript)

Reply:
I figured it out. The syntax is:
```suggestion
    def update_media(self, media: Union[Movie, Episode, Clip, Track]):
```

And of course, above you'd have to:
```python
from plexapi.audio import Track
from plexapi.video import Clip, Episode, Movie
```